scripts/start-server.sh

```diff
@@ -1,37 +1,12 @@
 #!/bin/bash
 
-. scripts/build-keycloak.sh
+set -v 
 
-set -v
+# Pull the docker image with the latest changes on Keycloak
+docker pull jboss/keycloak:master
 
-function waitForServer {
-  # Give the server some time to start up. Look for a well-known
-  # bit of text in the log file. Try at most 50 times before giving up.
-  C=50
-  while [ $C -gt 0 ]
-  do
-    grep "Keycloak ${VERSION} (WildFly Core .*) started" keycloak.log
-    if [ $? -eq 0 ]; then
-      echo "Server started."
-      C=0
-    else
-      echo -n "."
-      C=$(( $C - 1 ))
-    fi
-    sleep 1
-  done
-}
+# Start the image with default user/admin and import a default realm for integration tests
+docker run -d --name keycloak -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin -p 8080:8080 -v `pwd`/test/fixtures/auth-utils/nodejs-test-realm.json:/config/nodejs-test-realm.json -it jboss/keycloak:master -b 0.0.0.0  -Dkeycloak.migration.action=import -Dkeycloak.migration.provider=singleFile -Dkeycloak.migration.file=/config/nodejs-test-realm.json -Dkeycloak.migration.strategy=OVERWRITE_EXISTING
 
-KEYCLOAK="keycloak-server"
-
-# Start the server
-$KEYCLOAK/bin/add-user-keycloak.sh -u admin -p admin
-
-# Start the server
-$KEYCLOAK/bin/standalone.sh -Djava.net.preferIPv4Stack=true \
-                            -Dkeycloak.migration.action=import \
-                            -Dkeycloak.migration.provider=singleFile \
-                            -Dkeycloak.migration.file=./test/fixtures/auth-utils/nodejs-test-realm.json \
-                            -Dkeycloak.migration.strategy=OVERWRITE_EXISTING > keycloak.log 2>&1 &
-
-waitForServer
+# For debugging purposes to make sure the image was started
+docker ps -a
```

Comment:
Perhaps it would be better to add a block tries to fetch a URL from Keycloak to make sure it's up? It should do it in a loop for X seconds until it's up.

Reply:
Humm great idea.